resources/js/screens/DataEntry.vue

```diff
@@ -491,6 +496,14 @@
         this.showPlotForm = false
       },
     },
+
+    beforeRouteLeave(to, from, next) {
+      if (this.formsSet > 0) {
+        window.confirm('Do you really want to leave? You have unsaved changes!') ? next() : next(false)
```

Comment:
Instead of window.confirm, we can use the prettier this.$confirm()

```suggestion
        this.$confirm({
            title: 'Unsaved Changes!',
            text: 'Do you really want to leave? You have unsaved changes!',
            onConfirm: next,
        })
```

Reply:
I'll commit this so I can test it